### PR TITLE
[AIRFLOW-4227] Use python3-style type annotations.

### PR DESCRIPTION
**Pull request by jmcarp**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4227
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: run_unit_tests.sh [hunk 1]
```diff
@@ -34,9 +34,9 @@ nose_args=$@
 # Generate the `airflow` executable if needed
 which airflow > /dev/null || python setup.py develop
 
-echo "Initializing the DB"
```
Comment (BasPH) [inline run_unit_tests.sh]: Accident?

**Comment by BasPH** [comment]
Note: there is also this issue https://issues.apache.org/jira/browse/AIRFLOW-4205, part of a collection of issues on removing Python 2.

**Comment by jmcarp** [comment]
Thanks @BasPH. Also, will we support python3.5 in the future? If so, we can use function annotations but not variable annotations.

**Comment by ashb** [comment]
We should probably support  Python 3.5.2 which  is what Ubuntu ships

**Comment by jmcarp** [comment]
Thanks @ashb, I backed out variable annotations, and tests are passing on 3.5.x.

**Comment by mik-laj** [comment]
What do you think about replacing `collections.namedtuple` with `typing.NamedTuple`?
https://docs.python.org/3/library/typing.html#typing.NamedTuple

**Comment by jmcarp** [comment]
Added some typed namedtuples. Since we're supporting python3.5, we can't use variable annotations, so I used the backwards-compatible syntax.

**Comment by Fokko** [comment]
Can you rebase onto master?

**Comment by jmcarp** [comment]
Looks like #5090 will fix the kubernetes tests here once it's ready.

**Comment by jmcarp** [comment]
This should be ready to go now that #5090 is finished. cc @Fokko @kaxil

File: tests/plugins/test_plugin.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ class AirflowTestPlugin(AirflowPlugin):
     flask_blueprints = [bp]
     appbuilder_views = [v_appbuilder_package]
     appbuilder_menu_items = [appbuilder_mitem]
-    stat_name_handler = staticmethod(stat_name_dummy_handler)
+    stat_name_handler = stat_name_dummy_handler
```
Comment (Fokko) [inline tests/plugins/test_plugin.py:116]: This change seems unrelated?

**Review by Fokko — COMMENTED** [review]
Looking good @jmcarp 

One more comment, apart from that LGTM. Makes the code much more readable.

File: tests/plugins/test_plugin.py (hunk 2 above)
Comment (ashb) [reply tests/plugins/test_plugin.py:116]: It could be because mypy was complaining about this otherwise?

**Comment by feng-tao** [comment]
We could revisit this pr given we drop PY2?

File: tests/plugins/test_plugin.py (hunk 2 above)
Comment (jmcarp) [reply tests/plugins/test_plugin.py:116]: That's right @ashb, and mypy might be right to complain about `staticmethod` here, since `stat_name_dummy_handler` is a function and not a method, and already has the expected signature.

**Comment by mik-laj** [comment]
```bash
02:01 $ grep -RE 'namedtuple\(' . --include \*.py
./tests/models/test_connection.py:ConnectionParts = namedtuple("ConnectionParts", ["conn_type", "login", "password", "host", "port", "schema"])
./tests/contrib/hooks/test_azure_container_instance_hook.py:        named_instance = namedtuple("InstanceView", instance_view.keys())(*instance_view.values())
./tests/contrib/hooks/test_azure_container_instance_hook.py:        named_instance = namedtuple("Events", instance_view.keys())(*instance_view.values())
./tests/contrib/hooks/test_wasb_hook.py:        Blob = namedtuple('Blob', ['name'])
```
It seems to me that all classes have not been changed. WDYT?

**Comment by jmcarp** [comment]
@mik-laj: I hadn't changed all the named tuples in tests, but now I've gone through and dropped all uses of `collections.namedtuple`. Thanks for checking.

**Comment by jmcarp** [comment]
Looks like this was reimplemented in #5327. Thanks @BasPH.
